tests/utils.py

```diff
@@ -574,7 +576,7 @@ def pop_traces(self):
         return traces
 
 
-class DummyWriter(DummyWriterMixin, AgentWriter):
+class DummyWriter(DummyWriterMixin, AgentWriterInterface):
```

Comment:
feels like we are straying very far away from the original purpose of the DummyWriter

I wonder if just need to migrate all DummyTracer/DummyWriter usage to the test agent?

How many tests/places were you having issues with this? Low enough that we could try to convert some tests over now?

Reply:
It almost feels like this should just be `class DummyWriter(DummyWriterMixin, NativeWriter):` to remove a ton of that boilerplate work.